core/templates/pages/community-dashboard-page/translation-language-selector/translation-language-selector.component.ts

```diff
@@ -0,0 +1,114 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the translation language select.
+ */
+
+import {
+  Component, OnInit, Input, Output, EventEmitter, HostListener, ViewChild
+} from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+import { ContributionOpportunitiesBackendApiService } from
+  // eslint-disable-next-line max-len
+  'pages/community-dashboard-page/services/contribution-opportunities-backend-api.service';
+import { FeaturedTranslationLanguage } from
+  'domain/opportunity/FeaturedTranslationLanguageObjectFactory';
+import { LanguageUtilService } from 'domain/utilities/language-util.service';
+
+@Component({
+  selector: 'translation-language-selector',
+  templateUrl: './translation-language-selector.component.html'
+})
+export class TranslationLanguageSelectorComponent implements OnInit {
+  @Input() value: string;
+  @Output() setValue: EventEmitter<string> = new EventEmitter();
+  @ViewChild('dropdown', {'static': false}) dropdownRef;
+
+  options: {id: string, description: string}[];
+  languageIdToDescription: {[id: string]: string} = {};
+  featuredLanguages: FeaturedTranslationLanguage[] = [];
+
+  dropdownShown = false;
+  explanationPopupShown = false;
+  explanationPopupPxOffsetY = 0;
+  explanationPopupContent = '';
+
+  constructor(
+    private contributionOpportunitiesBackendApiService:
+      ContributionOpportunitiesBackendApiService,
+    private languageUtilService: LanguageUtilService
+  ) {}
+
+  ngOnInit() {
+    this.options = this.languageUtilService
+      .getAllVoiceoverLanguageCodes().map(languageCode => {
+        const description = this.languageUtilService
+          .getAudioLanguageDescription(languageCode);
+        this.languageIdToDescription[languageCode] = description;
+        return { id: languageCode, description };
+      });
+
+    this.contributionOpportunitiesBackendApiService
+      .fetchFeaturedTranslationLanguages()
+      .then((featuredLanguages: FeaturedTranslationLanguage[]) => {
+        this.featuredLanguages = featuredLanguages;
+      });
+  }
+
+  toggleDropdown() {
+    this.dropdownShown = !this.dropdownShown;
+  }
+
+  selectOption(value: string) {
+    this.setValue.emit(value);
+    this.dropdownShown = false;
+  }
+
+  showExplanationPopup(index: number) {
+    // Align popup to mouse-overed info icon
+    this.explanationPopupPxOffsetY = 75 + 30 * index;
+    this.explanationPopupContent = (
+      this.featuredLanguages[index].explanation);
+    this.explanationPopupShown = true;
+  }
+
+  hideExplanationPopup() {
+    this.explanationPopupShown = false;
+  }
+
+  /**
+   * Close popup when outside elements are clicked
+   * @param event mouse click event
+   */
+  @HostListener('document:click', ['$event'])
+  onDocumentClick(event: MouseEvent): void {
+    const targetElement = event.target as HTMLElement;
```

Comment:
Also, should we only show the explanation until the user hovers over the `i` button? (not sure!)

Reply:
Yes, that is what is happening currently. onDocumentClick closes the dropdown